OctoKit/Issue.swift

```diff
@@ -3,34 +3,80 @@ import RequestKit
+    public var labelsURL: NSURL?
+    public var commentsURL: NSURL?
+    public var eventsURL: NSURL?
+    public var htmlURL: NSURL?
     public var number: Int?
+    public var state: IssueState?
     public var title: String?
     public var body: String?
-    public var state: String?
     public var user: User?
+    public var labels: [Label]?
     public var assignee: User?
-    public var createdAt: NSDate?
-    public var closedAt: NSDate?
-    public var updatedAt: NSDate?
+    public var milestone: Milestone?
+    public var locked: Bool?
+    public var numberOfComments: Int?
+    public var closureDate: NSDate?
+    public var creationDate: NSDate?
+    public var updateDate: NSDate?
+    public var closer: User?
     
     public init(_ json: [String: AnyObject]) {
         if let id = json["id"] as? Int {
             self.id = id
-            url = json["url"] as? String
+            if let urlString = json["url"] as? String, url = NSURL(string: urlString) {
+                self.url = url
+            }
+            if let urlString = json["repository_url"] as? String, url = NSURL(string: urlString) {
+                repositoryURL = url
+            }
+            if let urlString = json["labels_url"] as? String, url = NSURL(string: urlString) {
+                labelsURL = url
+            }
+            if let urlString = json["comments_url"] as? String, url = NSURL(string: urlString) {
+                commentsURL = url
+            }
+            if let urlString = json["events_url"] as? String, url = NSURL(string: urlString) {
+                eventsURL = url
+            }
+            if let urlString = json["html_url"] as? String, url = NSURL(string: urlString) {
+                htmlURL = url
+            }
             number = json["number"] as? Int
+            switch json["state"] as? String ?? "" {
+            case "open":
+                state = .Open
+            case "closed":
+                state = .Closed
+            default:
+                break
+            }
             title = json["title"] as? String
             body = json["body"] as? String
-            state = json["state"] as? String
             user = User(json["user"] as? [String: AnyObject] ?? [:])
+            if let labelDictionaries = json["labels"] as? [[String: AnyObject]] {
+                labels = labelDictionaries.flatMap { Label($0) }
+            }
             assignee = User(json["assignee"] as? [String: AnyObject] ?? [:])
-            createdAt = Time.rfc3339Date(json["created_at"] as? String)
-            closedAt = Time.rfc3339Date(json["closed_at"] as? String)
-            updatedAt = Time.rfc3339Date(json["updated_at"] as? String)
+            milestone = Milestone(json["milestone"] as? [String: AnyObject] ?? [:])
+            locked = json["locked"] as? Bool
+            numberOfComments = json["comments"] as? Int
+            closureDate = Time.rfc3339Date(json["closed_at"] as? String)
```

Comment:
I preferred the old naming `createdAt`, `closedAt` and `updatedAt` as it maps the JSON and users of the library know what and where to look for.